plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/fs/DBFFileSystemManager.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.model.fs;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.Log;
+import org.jkiss.dbeaver.model.auth.SMSessionContext;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+import org.jkiss.dbeaver.runtime.DBWorkbench;
+import org.jkiss.utils.CommonUtils;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.nio.file.spi.FileSystemProvider;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class DBFFileSystemManager {
+    private static final Log log = Log.getLog(DBFFileSystemManager.class);
+    private final Map<String, FileSystemProvider> nioFileSystems = new LinkedHashMap<>();
```

Comment:
This looks very strange. File systems need extra info (project, cloud config id, etc).
We create them without any of these parameters. How does it work?